### ISSUE TITLE
Arrow in the console brings wrong statement after session restart

### ISSUE THREAD

**Issue by david-cortes** [rstudio/rstudio#8391]
**Steps to reproduce:**
* Start a new R session.
* Write and execute this command in the console: `print("before restart")`
* Click `Session` -> `Restart R`.
* Write and execute this command in the console: `print("after restart")`.
* Click `Session` -> `Restart R`.
* Write and execute this command in the console: `print("after restart2")`.
* Focus on the console and press the up arrow.

**Expected behavior:** should bring up `print("after restart2")`.

**Actual behavior:** brings up `print("before restart")`

**Setup:** RStudio 1.4.953.

**Comment by ronblum** [comment]
Thanks @david-cortes ! I can reproduce this on MacOS 11.0.1 as well in RStudio Desktop 1.4.1032. This is not an issue in 1.3.1093, so this appears to be a regression.

**Comment by mikebessuille** [comment]
Are we losing all the history after the restart?  or are we just showing the wrong command?  If the command history is lost, we should try to fix this for 1.4

**Comment by kevinushey** [comment]
I think the issue here is a bit more general. It seems like, after a restart, the first-executed command is not added to the history. So, a similar example here would be:

1. Execute `print("About to restart!")`.
2. Restart the R session,
3. Execute `print("I restarted!")`,
4. Press Up to access command history.

The `print("I restarted!")` command is not entered into the history, even though the rest of the command history is preserved.
